tiledb/api/src/array/fragment_info.rs

```diff
@@ -0,0 +1,879 @@
+    }
+
+    pub fn non_empty_domain(&self) -> TileDBResult<NonEmptyDomain> {
+        let schema = self.info.schema(self.index)?;
+        let num_dims = schema.domain()?.ndim()?;
+
+        let mut ret = Vec::<TypedRange>::new();
+        for dimension in 0..num_dims {
+            let range = self.info.non_empty_domain(self.index, dimension)?;
+            ret.push(range)
+        }
+
+        Ok(ret)
+    }
+
+    pub fn num_mbrs(&self) -> TileDBResult<u64> {
+        self.info.num_mbrs(self.index)
+    }
+
+    pub fn mbr<T: Into<LookupKey> + Clone>(
+        &self,
+        mbr_idx: u32,
+    ) -> TileDBResult<MinimumBoundingRectangle> {
+        let schema = self.info.schema(self.index)?;
+        let num_dims = schema.domain()?.ndim()?;
+
+        let mut ret = Vec::<TypedRange>::new();
+        for dimension in 0..num_dims {
+            let range = self.info.mbr(self.index, mbr_idx, dimension)?;
+            ret.push(range)
+        }
+
+        Ok(ret)
+    }
+
+    pub fn timestamp_range(&self) -> TileDBResult<(u64, u64)> {
+        self.info.timestamp_range(self.index)
+    }
+
+    pub fn has_consolidated_metadata(&self) -> TileDBResult<bool> {
+        self.info.has_consolidated_metadata(self.index)
+    }
+
+    pub fn to_vacuum_uri(&self) -> TileDBResult<String> {
+        self.info.to_vacuum_uri(self.index)
+    }
+}
+
+#[derive(ContextBound)]
+pub struct FragmentInfoList<'ctx> {
+    #[base(ContextBound)]
+    info: FragmentInfoInternal<'ctx>,
+}
+
+impl<'ctx> FragmentInfoList<'ctx> {
+    pub fn config(&self) -> TileDBResult<Config> {
+        self.info.config()
+    }
+
+    pub fn load(&self) -> TileDBResult<()> {
+        self.info.load()
+    }
+
+    pub fn unconsolidated_metadata_num(&self) -> TileDBResult<u32> {
+        self.info.unconsolidated_metadata_num()
+    }
+
+    pub fn num_to_vacuum(&self) -> TileDBResult<u32> {
+        self.info.num_to_vacuum()
+    }
+
+    pub fn total_cell_count(&self) -> TileDBResult<u64> {
+        self.info.total_cell_count()
+    }
+
+    pub fn num_fragments(&self) -> TileDBResult<u32> {
+        self.info.num_fragments()
+    }
+
+    pub fn get_fragment(&self, index: u32) -> TileDBResult<FragmentInfo> {
+        if index >= self.num_fragments()? {
+            return Err(Error::InvalidIndex(index as usize));
+        }
+
+        Ok(FragmentInfo {
+            info: &self.info,
+            index,
+        })
+    }
+
+    pub fn iter<'info: 'ctx>(
+        &'info self,
+    ) -> TileDBResult<FragmentInfoListIterator<'ctx, 'info>> {
+        FragmentInfoListIterator::try_from(self)
+    }
+}
+
+pub struct FragmentInfoListIterator<'ctx, 'info> {
+    info: &'info FragmentInfoList<'ctx>,
+    num_fragments: u32,
+    index: u32,
+}
+
+impl<'ctx, 'info: 'ctx> Iterator for FragmentInfoListIterator<'ctx, 'info> {
+    type Item = FragmentInfo<'ctx, 'info>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.index >= self.num_fragments {
+            return None;
```

Comment:
Looks like you can `impl FusedIterator`
https://doc.rust-lang.org/std/iter/trait.FusedIterator.html

Reply:
Neat! Done.